dbms/src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp

```diff
@@ -227,9 +236,23 @@ bool JSONEachRowRowInputFormat::readRow(MutableColumns & columns, RowReadExtensi
     if (!in.eof() && (*in.position() == ',' || *in.position() == ';'))
         ++in.position();
 
+    /// Finish reading rows if data is in square brackets and ']' received.
     skipWhitespaceIfAny(in);
-    if (in.eof())
+    if (!in.eof() && *in.position() == ']' && data_in_square_brackets) {
+        data_in_square_brackets = false;
+        ++in.position();
+        skipWhitespaceIfAny(in);
+        if (in.eof())
+            return false;
+        throw Exception("Unexpected data after ']' while parsing JSONEachRow format.", ErrorCodes::INCORRECT_DATA);
```

Comment:
Just stop parsing is Ok. The user may continue to read from this ReadBuffer in another format.